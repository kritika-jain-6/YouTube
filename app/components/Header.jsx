import React, { Component } from "react";
import {
  StyleSheet,
  View,
  Image,
  TouchableOpacity,
  SafeAreaView,
} from "react-native";
import Feather from "@expo/vector-icons/Feather";
import { withNavigationHOC } from "../withNavigationHOC";
import { router } from "expo-router";

class Header extends Component {
  handleSearchPress = () => {
    this.props.navigation.navigate("SearchScreen");
  };

  handleBellPress = () => {
    this.props.navigation.navigate("NotificationScreen");
  };

  render() {
    const { onCastPress, navigation,route } = this.props;
    const isNotificationScreen = route?.name === "NotificationScreen";
    return (
      <SafeAreaView testID="header-container" style={styles.container}>
  
  {isNotificationScreen ? (
          <View style={styles.notificationHeader}>
            <TouchableOpacity onPress={() => navigation.back()}>
              <Ionicons name="arrow-back" size={24} color="black" />
            </TouchableOpacity>
            <Text style={styles.notificationText}>Notifications</Text>
          </View>
        ) : (
          <Image
            testID="yt-logo"
            style={styles.image}
            source={require("../assets/images/yt-logo.png")}
            resizeMode="contain"
          />
        )}

        <View style={styles.btns}>
          <TouchableOpacity
            testID="cast-btn"
            style={styles.btn}
            onPress={onCastPress}
          >
            <Feather name="cast" size={22} color={"#555"} />
          </TouchableOpacity>
          <TouchableOpacity
            testID="bell-btn"
            style={styles.btn}
            onPress={this.handleBellPress}
          >
            <Feather name="bell" size={22} color={"#555"} />
          </TouchableOpacity>
          <TouchableOpacity
            testID="search-btn"
            style={styles.btn}
            onPress={this.handleSearchPress}
          >
            <Feather name="search" size={22} color={"#555"} />
          </TouchableOpacity>
        </View>
      </SafeAreaView>
    );
  }
}

export default withNavigationHOC(Header);

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    width: "100%",
    paddingHorizontal: 15,
    backgroundColor: "#fff",
  },
 
  notificationHeader: {
    flexDirection: "row",
    alignItems: "center",
  },
  notificationText: {
    fontSize: 18,
    fontWeight: "bold",
    marginLeft: 10,
  },
  image: {
    width: 120,
    height: 50,
  },
  btns: {
    flexDirection: "row",
    alignItems: "center",
  },
  btn: {
    marginHorizontal: 10,
  },
});
