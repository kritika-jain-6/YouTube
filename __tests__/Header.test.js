import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Header from "../app/components/Header";  

jest.mock("../app/withNavigationHOC.jsx", () => ({
  withNavigationHOC: (component) => component, 
}));


describe("Header Component", () => {
  let mockNavigate, mockOnCastPress;

  beforeEach(() => {
    mockNavigate = jest.fn(); // Mock navigation function
    mockOnCastPress = jest.fn(); // Mock cast button function
  });


  it("renders the Header component", () => {
    const { getByTestId } = render(<Header navigation={{ navigate: mockNavigate }} />);
    
    expect(getByTestId("header-container")).toBeTruthy();
  });


  it("renders the YouTube logo", () => {
    const { getByTestId } = render(<Header navigation={{ navigate: mockNavigate }} />);
    
    expect(getByTestId("yt-logo")).toBeTruthy();
  });


  it("renders the Search button", () => {
    const { getByTestId } = render(<Header navigation={{ navigate: mockNavigate }} />);
    
    expect(getByTestId("search-btn")).toBeTruthy();
  });

 
  it("renders the Notification Bell button", () => {
    const { getByTestId } = render(<Header navigation={{ navigate: mockNavigate }} />);
    
    expect(getByTestId("bell-btn")).toBeTruthy();
  });


  it("renders the Cast button", () => {
    const { getByTestId } = render(<Header navigation={{ navigate: mockNavigate }} />);
    
    expect(getByTestId("cast-btn")).toBeTruthy();
  });


  it("navigates to SearchScreen when search button is pressed", () => {
    const { getByTestId } = render(<Header navigation={{ navigate: mockNavigate }} />);

    fireEvent.press(getByTestId("search-btn")); // Simulate button press

    expect(mockNavigate).toHaveBeenCalledWith("SearchScreen"); // Check navigation
  });

  
  it("navigates to NotificationScreen when bell button is pressed", () => {
    const { getByTestId } = render(<Header navigation={{ navigate: mockNavigate }} />);

    fireEvent.press(getByTestId("bell-btn"));

    expect(mockNavigate).toHaveBeenCalledWith("NotificationScreen");
  });

  it("calls the onCastPress prop when cast button is pressed", () => {
    const { getByTestId } = render(<Header navigation={{ navigate: mockNavigate }} onCastPress={mockOnCastPress} />);

    fireEvent.press(getByTestId("cast-btn"));

    expect(mockOnCastPress).toHaveBeenCalled(); // Check function call
  });

 
  it("renders Feather icons correctly", () => {
    const { getByTestId } = render(<Header navigation={{ navigate: mockNavigate }} />);

    expect(getByTestId("search-btn")).toBeTruthy();
    expect(getByTestId("bell-btn")).toBeTruthy();
    expect(getByTestId("cast-btn")).toBeTruthy();
  });


  it("applies correct styles", () => {
    const { getByTestId } = render(<Header navigation={{ navigate: mockNavigate }} />);
    const headerContainer = getByTestId("header-container");

    expect(headerContainer.props.style).toMatchObject({
      flexDirection: "row",
      justifyContent: "space-between",
      alignItems: "center",
      width: "100%",
      paddingHorizontal: 15,
      backgroundColor: "#fff",
    });
  });
});
