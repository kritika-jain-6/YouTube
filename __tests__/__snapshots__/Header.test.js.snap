// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches the snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 15,
      "width": "100%",
    }
  }
  testID="header-container"
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../app/assets/images/yt-logo.png",
      }
    }
    style={
      {
        "height": 60,
        "width": 120,
      }
    }
    testID="yt-logo"
  />
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginHorizontal": 10,
          "opacity": 1,
        }
      }
      testID="cast-btn"
    >
      <Feather
        color="#555"
        name="cast"
        size={22}
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginHorizontal": 10,
          "opacity": 1,
        }
      }
      testID="bell-btn"
    >
      <Feather
        color="#555"
        name="bell"
        size={22}
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginHorizontal": 10,
          "opacity": 1,
        }
      }
      testID="search-btn"
    >
      <Feather
        color="#555"
        name="search"
        size={22}
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
